src-tauri/src/node.rs

```diff
@@ -215,6 +214,9 @@ fn create_configuration<CS: ChainSpec + 'static>(
                 "https://tauri.localhost".to_string(),
             ]),
             rpc_max_payload: None,
+            rpc_max_request_size: None,
```

Comment:
@nazar-pc , what should be the value of these new fields?
Node is not responding to farmer at the moment. 
![image](https://user-images.githubusercontent.com/32795992/169311663-fbf69c6a-804e-441b-b46b-4c505e31a62d.png)
This issue is not present in the main branch and farmer works. Could it be related to here?

Reply:
I'm referring to: line 218, 219 and 239

Reply:
i can put `Default::default()` as well

Reply:
Tried them, also tried `rpc_max_subs_per_conn: Some(1024),`
did not help. Maybe the bug is not related to here. 

But this bug is introduced with this PR, so it has to be related to some of the code changes...

Reply:
Can you connect to node with Polkadot.js apps?

Reply:
BTW, farmer seems to also have issues, what is in its logs?

Reply:
Polkadot.js also cannot connect to the node. That's why, my assumption is the following: node runs, but does not successfully listens on 9944.

I will send farmer logs soon

Reply:
For farmer nodes, i will change the frontend a bit via bypassing the polkadot.js api connection from the frontend. 
In the logs above I send: it doesn't even starts the farmer. Because the logic is the following in the frontend:
1. start the node
2. connect to the node (FAILS)
3. start the farmer (does not even triggered, since step 2 is failed)

Reply:
I bypassed the connection to node from frontend. Farmer started and immediately crashed:
```
[2022-05-19][20:06:12][subspace_desktop::farmer][INFO] Connecting to node at ws://127.0.0.1:9944
[2022-05-19][20:06:12][soketto::connection][WARN] da091994: accumulated message length exceeds maximum

====================

Version: 0.5.0-74fa368447e

   0: backtrace::backtrace::libunwind::trace
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.64/src/backtrace/libunwind.rs:93:5
      backtrace::backtrace::trace_unsynchronized
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.64/src/backtrace/mod.rs:66:5
      backtrace::backtrace::trace
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.64/src/backtrace/mod.rs:53:14
   1: backtrace::capture::Backtrace::create
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.64/src/capture.rs:176:9
      backtrace::capture::Backtrace::new
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.64/src/capture.rs:140:22
   2: sp_panic_handler::panic_hook
             at /Users/ozgunozerk/.cargo/git/checkouts/substrate-7e08433d4c370a21/5d3e7c4/primitives/panic-handler/src/lib.rs:166:18
      sp_panic_handler::set::{{closure}}
             at /Users/ozgunozerk/.cargo/git/checkouts/substrate-7e08433d4c370a21/5d3e7c4/primitives/panic-handler/src/lib.rs:62:12
   3: std::panicking::rust_panic_with_hook
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/panicking.rs:702:17
   4: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/panicking.rs:588:13
   5: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/sys_common/backtrace.rs:138:18
   6: rust_begin_unwind
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/panicking.rs:584:5
   7: core::panicking::panic_fmt
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/core/src/panicking.rs:143:14
   8: core::result::unwrap_failed
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/core/src/result.rs:1749:5
   9: core::result::Result<T,E>::unwrap
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/core/src/result.rs:1065:23
      subspace_desktop::farming::{{closure}}
             at src/main.rs:49:9
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/core/src/future/mod.rs:84:19
  10: <futures_util::future::future::map::Map<Fut,F> as core::future::future::Future>::poll
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.21/src/future/future/map.rs:55:37
  11: <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.21/src/lib.rs:91:13
      subspace_desktop::main::{{closure}}::{{closure}}::{{closure}}
             at src/main.rs:126:13
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/core/src/future/mod.rs:84:19
      tauri::hooks::InvokeResolver<R>::respond_async_serialized::{{closure}}
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tauri-1.0.0-rc.11/src/hooks.rs:196:44
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/core/src/future/mod.rs:84:19
  12: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/core.rs:165:17
      tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/loom/std/unsafe_cell.rs:14:9
      tokio::runtime::task::core::CoreStage<T>::poll
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/core.rs:155:13
      tokio::runtime::task::harness::poll_future::{{closure}}
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/harness.rs:470:19
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/panicking.rs:492:40
      std::panicking::try
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/panicking.rs:456:19
      std::panic::catch_unwind
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/panic.rs:137:14
      tokio::runtime::task::harness::poll_future
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/harness.rs:458:18
      tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/harness.rs:104:27
      tokio::runtime::task::harness::Harness<T,S>::poll
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/harness.rs:57:15
  13: tokio::runtime::task::raw::RawTask::poll
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/raw.rs:84:18
      tokio::runtime::task::LocalNotified<S>::run
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/mod.rs:376:9
      tokio::runtime::thread_pool::worker::Context::run_task::{{closure}}
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/thread_pool/worker.rs:454:21
      tokio::coop::with_budget::{{closure}}
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/coop.rs:102:9
      std::thread::local::LocalKey<T>::try_with
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/thread/local.rs:413:16
      std::thread::local::LocalKey<T>::with
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/thread/local.rs:389:9
  14: tokio::coop::with_budget
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/coop.rs:95:5
      tokio::coop::budget
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/coop.rs:72:5
      tokio::runtime::thread_pool::worker::Context::run_task
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/thread_pool/worker.rs:430:9
  15: tokio::runtime::thread_pool::worker::Context::run
  16: tokio::runtime::thread_pool::worker::run::{{closure}}
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/thread_pool/worker.rs:382:17
      tokio::macros::scoped_tls::ScopedKey<T>::set
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/macros/scoped_tls.rs:61:9
  17: tokio::runtime::thread_pool::worker::run
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/thread_pool/worker.rs:379:5
  18: tokio::runtime::thread_pool::worker::Launch::launch::{{closure}}
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/thread_pool/worker.rs:358:45
      <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/blocking/task.rs:42:21
  19: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/core.rs:165:17
      tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/loom/std/unsafe_cell.rs:14:9
      tokio::runtime::task::core::CoreStage<T>::poll
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/core.rs:155:13
      tokio::runtime::task::harness::poll_future::{{closure}}
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/harness.rs:470:19
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/panicking.rs:492:40
      std::panicking::try
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/panicking.rs:456:19
      std::panic::catch_unwind
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/panic.rs:137:14
      tokio::runtime::task::harness::poll_future
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/harness.rs:458:18
      tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/harness.rs:104:27
      tokio::runtime::task::harness::Harness<T,S>::poll
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/harness.rs:57:15
  20: tokio::runtime::task::raw::RawTask::poll
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/raw.rs:84:18
      tokio::runtime::task::UnownedTask<S>::run
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/task/mod.rs:413:9
      tokio::runtime::blocking::pool::Task::run
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/blocking/pool.rs:91:9
      tokio::runtime::blocking::pool::Inner::run
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/blocking/pool.rs:325:17
  21: tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}
             at /Users/ozgunozerk/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/blocking/pool.rs:300:13
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/sys_common/backtrace.rs:122:18
  22: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/thread/mod.rs:498:17
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/panicking.rs:492:40
      std::panicking::try
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/panicking.rs:456:19
      std::panic::catch_unwind
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/panic.rs:137:14
      std::thread::Builder::spawn_unchecked_::{{closure}}
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/thread/mod.rs:497:30
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/core/src/ops/function.rs:227:5
  23: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/alloc/src/boxed.rs:1854:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/alloc/src/boxed.rs:1854:9
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/c5c610aad0a012a9228ecb83cc19e77111a52140/library/std/src/sys/unix/thread.rs:108:17
  24: __pthread_deallocate


Thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: The background task been terminated because: Invalid request ID; restart required', src/main.rs:51
```

Reply:
Looks like missing limit, where instead of `None` there should have been `Some(15728640)`

Reply:
Found the bug. After applying `Default::default()` to the new node config parameters, and bypassing the frontend connections to the node, farmer finally worked! 

At the moment, now we know 3 things for sure:
- node works
- farmer works
- frontend cannot connect to node

Reply:
Frontend not being able to connect to node can be because of 2 things:
- there is something wrong in the frontend
- there is something wrong in the node (blocking connections from frontend)

I'm inclined to think, it is the second one. Since we haven't changed the frontend at all. And everything is working in the main branch. @isSerge , we have to debug is further. This issue is present in both DEV and PROD, we should be able to debug it quickly after this point.

Reply:
`Default::default()` for `Option` is `None`, so that change was unnecessary.

Reply:
Found the bug of frontend as well, and it is ridiculous.
the list of allowed connections for the node configuration:
```rust
rpc_cors: Some(vec![
                "http://localhost:*".to_string(),
                "http://127.0.0.1:*".to_string(),
                "https://localhost:*".to_string(),
                "https://127.0.0.1:*".to_string(),
                "https://polkadot.js.org".to_string(),
                "tauri://localhost".to_string(),
                "https://tauri.localhost".to_string(),
                "http://localhost:8080".to_string(), // WE HAD TO ADD THIS
            ]),
```

How come `"http://localhost:*".to_string(),` does not include `"http://localhost:8080".to_string()` anymore? Should I open an issue to substrate about this?

Reply:
Yeah, this looks like a breaking change in behavior to me.